### PR TITLE
SLING-11571 repoinit: allow add or remove mixin types

### PR DESCRIPTION
**Pull request by enapps-enorman**

### CODE REVIEW

File: src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    @Test
+    public void addMixinOnNotExistingPath() throws Exception {
+        // this should just log a warning and continue
+        U.parseAndExecute("add mixin mix:lockable to /addMixinOnNotExistingPath");
+        assertNodeNotExists("/addMixinOnNotExistingPath");
+    }
+
+    @Test
+    public void removeMixinFromNotExistingPath() throws Exception {
+        // this should just log a warning and continue
```
Comment (bdelacretaz) [inline src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java:101]: FWIW you could use a LogCapture utility to verify logging, as done in https://github.com/apache/sling-org-apache-sling-graphql-core/blob/d430d47df5b190107d413f31ef19052aacab4d98/src/test/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutorLoggingTest.java

File: src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    public void removeTwoMixinsFromTwoPaths() throws Exception {
+        U.parseAndExecute("create path /removeTwoMixinsOnOnePath1(nt:unstructured mixin mix:lockable,mix:referenceable)\n"
+                + "create path /removeTwoMixinsOnOnePath2(nt:unstructured mixin mix:lockable,mix:referenceable)");
+        U.assertNodeExists("/removeTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+        U.assertNodeExists("/removeTwoMixinsOnOnePath2", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+
+        U.parseAndExecute("remove mixin mix:lockable,mix:referenceable from /removeTwoMixinsOnOnePath1,/removeTwoMixinsOnOnePath2");
```
Comment (bdelacretaz) [inline src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java:87]: Same comment as above about not testing selective removal of mixins

File: src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+    }
+
+    @Test
+    public void removeOneMixinFromOnePath() throws Exception {
+        U.parseAndExecute("create path /removeOneMixinOnOnePath(nt:unstructured mixin mix:lockable)");
+        U.assertNodeExists("/removeOneMixinOnOnePath", "nt:unstructured", Collections.singletonList("mix:lockable"));
+
+        U.parseAndExecute("remove mixin mix:lockable from /removeOneMixinOnOnePath");
```
[the remark this answers is not in the document]
Comment (enapps-enorman) [reply src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java:76]: Thanks for the review.  I've incorporated the suggested change at: https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/38

File: src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java (hunk 2 above)
Comment (enapps-enorman) [reply src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java:87]: Thanks for the review.  I've incorporated the suggested change at: https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/38

File: src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java (hunk 1 above)
Comment (enapps-enorman) [reply src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java:101]: Thanks for the review.  I've incorporated the suggested change at: https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/38

**Pull request merged**